app/models/test_result.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+# Representation of the TestResult XML property in an OpenSCAP report. Holds the
+# basic report properties, such as dates, host, and results.
+class TestResult < ApplicationRecord
+  belongs_to :profile
+  belongs_to :host
+  has_one :benchmark, through: :profile
+  has_many :rule_results, dependent: :delete_all
+
+  validates :host_id, presence: true
+  validates :profile_id, presence: true
```

Comment:
What's the uniqueness validation here? We should define it and also add a DB constraint or index.